src/main/java/io/harness/cf/client/connector/LocalConnector.java

```diff
@@ -0,0 +1,165 @@
+package io.harness.cf.client.connector;
+
+import com.google.gson.Gson;
+import io.harness.cf.model.FeatureConfig;
+import io.harness.cf.model.Metrics;
+import io.harness.cf.model.Segment;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import lombok.NonNull;
+
+public class LocalConnector implements Connector {
+  private final String source;
+  private final Gson gson = new Gson();
+  private FileWatcher flagsWatcher;
+  private FileWatcher segmentsWatcher;
+
+  private final ExecutorService pool = Executors.newFixedThreadPool(2);
+
+  public LocalConnector(@NonNull String source) {
+    this.source = source;
+  }
+
+  @Override
+  public Optional<String> authenticate(Consumer<String> onError) {
+    return Optional.of("success");
+  }
+
+  @Override
+  public List<FeatureConfig> getFlags() {
+    List<FeatureConfig> configs = new ArrayList<>();
+    try {
+      List<File> files =
+          Files.list(Paths.get(source, "flags"))
+              .filter(Files::isRegularFile)
+              .filter(path -> path.toString().endsWith(".json"))
+              .map(Path::toFile)
+              .collect(Collectors.toList());
+
+      files.forEach(
+          file -> {
+            try {
+              String content = new String(Files.readAllBytes(file.toPath()));
+              FeatureConfig featureConfig = gson.fromJson(content, FeatureConfig.class);
+              if (featureConfig != null) {
+                configs.add(featureConfig);
+              }
+            } catch (Exception e) {
+              e.printStackTrace();
+            }
+          });
```

Comment:
Is it worth extracting these instead of continuing to precess in the stream? Doesn't look like there is any special handling of the exception. You could also extract the bit that creates the FeatureConfig to a method with try/catch, and still use it in the stream.